presto-resource-group-managers/src/test/java/com/facebook/presto/resourceGroups/db/TestDbResourceGroupConfigurationManager.java

```diff
@@ -285,6 +286,67 @@ public void testSelectorPriority()
         }
     }
 
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "No selectors are configured")
+    public void testInvalidConfiguration()
+    {
+        H2DaoProvider daoProvider = setup("selectors");
+        H2ResourceGroupsDao dao = daoProvider.get();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "100%", 100, 100, 100, null, null, null, null, null, null, null, null, ENVIRONMENT);
+
+        DbResourceGroupConfigurationManager manager = new DbResourceGroupConfigurationManager(
+                (poolId, listener) -> {},
+                new DbResourceGroupConfig().setMaxRefreshInterval(Duration.valueOf("1ms")),
+                daoProvider.get(),
+                ENVIRONMENT);
+
+        manager.load();
+
+        manager.getSelectors();
+    }
+
+    @Test
+    public void testRefreshInterval()
+            throws Exception
+    {
+        H2DaoProvider daoProvider = setup("selectors");
+        H2ResourceGroupsDao dao = daoProvider.get();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "100%", 100, 100, 100, null, null, null, null, null, null, null, null, ENVIRONMENT);
+
+        DbResourceGroupConfigurationManager manager = new DbResourceGroupConfigurationManager(
+                (poolId, listener) -> {},
+                new DbResourceGroupConfig().setMaxRefreshInterval(Duration.valueOf("1ms")),
+                daoProvider.get(),
+                ENVIRONMENT);
+
+        manager.start();
```

Comment:
In this test I guess what you are trying to do is:
1) schedule the thread to load the config with `start()` 
2) drop the selectors table
3) wait for that thread to reload the config with the sleeps

First of all I think this test is flaky, because it's possible that the configuration loading thread doesn't get scheduled in the next 3+ seconds (and doesn't reload the config, so selectors is empty at that point) and this test will fail.

Instead of relying on the config loader thread and depending on sleeps, I think we can do:

```
DbResourceGroupConfigurationManager manager = new DbResourceGroupConfigurationManager(
        (poolId, listener) -> {},
        new DbResourceGroupConfig().setMaxRefreshInterval(Duration.valueOf("1ms")),
        daoProvider.get(),
        ENVIRONMENT);
// the constructor will load the config
dao.dropSelectorsTable();
// reload the config
manager.load();

//asserts ...
```